beacon-chain/db/filesystem/data_column.go

```diff
@@ -0,0 +1,966 @@
+		if len(dataColumnSidecars) == 0 {
+			break
+		}
+
+		for _, dataColumnSidecar := range dataColumnSidecars {
+			// Extract the data columns index.
+			dataColumnIndex := dataColumnSidecar.ColumnIndex
+
+			// Skip if the data column is already saved.
+			if metadata.indices[dataColumnIndex] >= limit {
+				continue
+			}
+
+			// Check if the number of saved data columns is too large.
+			// This is impossible to happen in practice is this function is called
+			// by SaveDataColumnSidecars.
+			if metadata.savedDataColumnSidecarCount >= mandatoryNumberOfColumns {
+				return ErrTooManyDataColumns
+			}
+
+			// SSZ encode the data column sidecar.
+			sszEncodedDataColumnSidecar, err := dataColumnSidecar.MarshalSSZ()
+			if err != nil {
+				return errors.Wrap(err, "data column sidecar marshal SSZ")
+			}
+
+			// Compute the size of the SSZ encoded data column sidecar.
+			incomingSszEncodedDataColumnSidecarSize := uint32(len(sszEncodedDataColumnSidecar))
+
+			// Check if the incoming encoded data column sidecar size corresponds to the one read from the file.
+			if incomingSszEncodedDataColumnSidecarSize != metadata.sszEncodedDataColumnSidecarSize {
+				return ErrWrongSszEncodedDataColumnSidecarSize
+			}
+
+			// Alter indices to mark the data column as saved.
+			// savedDataColumnsCount can safely be cast to uint8 since we have checked it is less than mandatoryNumberOfColumns.
+			metadata.indices[dataColumnIndex] = limit + uint8(metadata.savedDataColumnSidecarCount)
+			metadata.savedDataColumnSidecarCount++
+
+			// Append the SSZ encoded data column sidecar to the SSZ encoded data column sidecars.
+			sszEncodedDataColumnSidecars = append(sszEncodedDataColumnSidecars, sszEncodedDataColumnSidecar...)
+		}
+	}
+
+	// Save indices to the file.
+	count, err := file.WriteAt(metadata.indices[:], int64(versionSize+encodedSszEncodedDataColumnSidecarSizeSize))
+	if err != nil {
+		return errors.Wrap(err, "write indices")
+	}
+	if count != mandatoryNumberOfColumns {
+		return ErrWrongBytesWritten
+	}
+
+	// Append the SSZ encoded data column sidecars to the end of the file.
+	count, err = file.WriteAt(sszEncodedDataColumnSidecars, metadata.fileSize)
+	if err != nil {
+		return errors.Wrap(err, "write SSZ encoded data column sidecars")
+	}
+	if count != len(sszEncodedDataColumnSidecars) {
+		return ErrWrongBytesWritten
+	}
+
+	return nil
+}
+
+// saveDataColumnSidecarsNewFile saves data column sidecars into a new file.
+// This function expects all data column sidecars to belong to the same block.
+func (dcs *DataColumnStorage) saveDataColumnSidecarsNewFile(filePath string, inputDataColumnSidecars chan []blocks.VerifiedRODataColumn) (err error) {
+	// Initialize the indices.
+	var indices [mandatoryNumberOfColumns]byte
+
+	var (
+		sszEncodedDataColumnSidecarRefSize int
+		sszEncodedDataColumnSidecars       []byte
+	)
+
+	// Initialize the count of the saved SSZ encoded data column sidecar.
+	storedCount := 0
+
+	log.Error("AAAA")
```

Comment:
AAAA!

Reply:
Fixed in https://github.com/prysmaticlabs/prysm/pull/15061/commits/6d16ed4a89504b2c7c067f5a15ea7bd7b65b049b.